### DISCUSSION TITLE
Option to ignore `Visit` nodes based on comments.

### DISCUSSION THREAD

**Discussion by flickyiyo** [swc-project/swc#7903] [Q&A] [answered]
Currently I am working on a tool to find unused exports using `Visit` trait implementations, problem being that some exports need to be preserved even when they are not used outside the file they are declared. I've worked creating custom eslint rules before and there is a way to disable eslint during codescan in specific files, line, or blocks of code using comments such as `/* eslint-disable */`.

Is there any feature here that I can use create such functionality? 
If not, where/how can I get started to add said option in swc?

**Comment by kwonoj** [comment]
If you're implementing a visitor in a specific node you could check attached comments to do necessary conditions. There are some refereces around existing swc plugins out there. If you're asking to do have some feature supported out of the box by `@swc/core`, the answer is probably no.

**Reply by flickyiyo** [reply, marked as answer]
After looking at the documentation, `swc_core::common::comments::SingleThreadedComments` provided the functionality I needed to implement a webpack's magic comment like feature in my project.
